modules/cuml/src/node_cuml/umap.hpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2021, NVIDIA CORPORATION.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include "cuml/manifold/umap.hpp"
+#include "cuml/manifold/umapparams.hpp"
+#include "raft/handle.hpp"
+
+#include <node_cudf/column.hpp>
+
+#include <nv_node/objectwrap.hpp>
+#include <nv_node/utilities/args.hpp>
+
+#include <cuml/common/device_buffer.hpp>
+
+#include <napi.h>
+
+namespace nv {
+/**
+ * @brief An owning wrapper around a cuml::manifold::UMAP
+ *
+ */
+struct UMAP : public EnvLocalObjectWrap<UMAP> {
+  /**
+   * @brief Initialize and export the UMAP JavaScript constructor and prototype
+   *
+   * @param env The active JavaScript environment
+   * @param exports The exports object to decorate
+   * @return Napi::Function The UMAP constructor function
+   */
+  static Napi::Function Init(Napi::Env const& env, Napi::Object exports);
+
+  /**
+   * @brief Construct a new UMAP instance from C++.
+   */
+  static wrapper_t New(Napi::Env const& env);
+
+  /**
+   * @brief Construct a new UMAP instance from JavaScript
+   *
+   * @param args
+   */
+  UMAP(CallbackArgs const& args);
+
+  void fit(float* X,
+           cudf::size_type n_samples,
+           cudf::size_type n_features,
+           float* y,
+           int64_t* knn_indices,
+           float* knn_dists,
+           bool convert_dtype,
+           float* embeddings);
```

Comment:
We shouldn't have public methods that take raw pointers, use const refs to definitely-initialized types instead:

```suggestion
  void fit(DeviceBuffer::wrapper_t const& X,
           cudf::size_type n_samples,
           cudf::size_type n_features,
           DeviceBuffer::wrapper_t const& y,
           DeviceBuffer::wrapper_t const& knn_indices,
           DeviceBuffer::wrapper_t const& knn_dists,
           bool convert_dtype,
           DeviceBuffer::wrapper_t const& embeddings);
```